ansible/roles/vm/install/tasks/main.yml

```diff
@@ -1,8 +1,49 @@
 ---
-- name: generate preseed file
-  template:
-    src: "preseed_{{ vmdistro }}-{{ vmdistcodename }}.cfg.j2"
-    dest: "{{ vm_host.installer.preseed_path }}/vm-{{ vmname }}-{{ vmdistro }}-{{ vmdistcodename }}.cfg"
+- block:
+    - name: Make a temporary directory
+      command: mktemp -d
+      register: tmpdir
+
+    - set_fact: tmpdir={{ tmpdir.stdout }}
+
+    - name: generate preseed file
+      template:
+        src: "preseed_{{ vmdistro }}-{{ vmdistcodename }}.cfg.j2"
+        dest: "{{ tmpdir }}/preseed.cfg"
+
+    - name: Extract the initramfs
+      shell: gunzip -c "{{ vm_host.installer.path }}/{{ vmdistro }}-{{ vmdistcodename }}/{{ hostvars[vmname].vm_install_cooked.arch | default('amd64') }}/initrd.gz" > "{{ tmpdir }}/initrd"
+
```

Comment:
Actually unpacking the initrd.gz is not necessary. See:
https://unix.stackexchange.com/questions/243657/appending-files-to-initramfs-image-reliable

The append only method has the benefit that it works even if your are not root.

Reply:
Oh, right, gzip is appendable.

Reply:
PS: This thing should have worked as not-root, though, as I'm wasn't modifying the original, but I agree that appending directly to (a copy of) the original file is much better.